style_editor.html

```diff
@@ -9456,37 +9456,31 @@
           option.text = nameWithoutEffect;
 
           // Check if the effect belongs to a certain category and add it to the corresponding sub-group
-          if (name.startsWith('EFFECT_GAME')) {
-            gameEffectsMenu.appendChild(option);
-          } else if (name.startsWith('EFFECT_USER')) {
-            userEffectsMenu.appendChild(option);
-          } else if ([
-              EFFECT_STUN, 
-              EFFECT_FIRE, 
-              EFFECT_CLIP_IN, 
-              EFFECT_CLIP_OUT, 
-              EFFECT_RELOAD, 
-              EFFECT_MODE, 
-              EFFECT_RANGE, 
-              EFFECT_EMPTY, 
-              EFFECT_FULL, 
-              EFFECT_JAM, 
-              EFFECT_UNJAM, 
-              EFFECT_PLI_ON, 
-              EFFECT_PLI_OFF
-            ].includes(Number(value))) {
-            blasterEffectsMenu.appendChild(option);
-          } else if ([
-            EFFECT_ERROR_IN_BLADE_ARRAY,
-            EFFECT_ERROR_IN_FONT_DIRECTORY,
-            EFFECT_FONT_DIRECTORY_NOT_FOUND,
-            EFFECT_SD_CARD_NOT_FOUND,
-            EFFECT_LOW_BATTERY
-            ].includes(Number(value))) {
-            errorMessagesMenu.appendChild(option);
-          } else {
-            generalEffectsMenu.appendChild(option);
+          switch (true) {
+            case name.startsWith('EFFECT_GAME'):
```

Comment:
does this work?


I was thinking something like this:

if (name.startswith(....)) {
} else if (name.startswith(....)) {
} else {
   switch (Number(value)) {
      case EFFECT_STUN:
      case EFFECT_FIRE:
       .....
           blasterEffectsMenu.appendChild(option);
           break;
      case EFFECT_ERROR_IN_BLADE_ARRAY:
       ...
           errorMessagesMenu.appendChild(option);
           break;
      default:
           generalEffectsMenu.appendChild(option);
           break;
     }
 }

Reply:
It does work. Apparently JS lets you use an array directly in a case statement. The `includes` method checks the array and returns T or F .
Of course if you prefer,  it can be broken up one case per effect.
```
if (name.startsWith('EFFECT_GAME')) {
  gameEffectsMenu.appendChild(option);
} else if (name.startsWith('EFFECT_USER')) {
  userEffectsMenu.appendChild(option);
} else {
  switch (Number(value)) {
    case EFFECT_STUN:
    case EFFECT_FIRE:
    case EFFECT_CLIP_IN:
    case EFFECT_CLIP_OUT:
    case EFFECT_RELOAD:
    case EFFECT_MODE:
    case EFFECT_RANGE:
    case EFFECT_EMPTY:
    case EFFECT_FULL:
    case EFFECT_JAM:
    case EFFECT_UNJAM:
    case EFFECT_PLI_ON:
    case EFFECT_PLI_OFF:
      blasterEffectsMenu.appendChild(option);
      break;
    case EFFECT_ERROR_IN_BLADE_ARRAY:
    case EFFECT_ERROR_IN_FONT_DIRECTORY:
    case EFFECT_FONT_DIRECTORY_NOT_FOUND:
    case EFFECT_SD_CARD_NOT_FOUND:
    case EFFECT_LOW_BATTERY:
      errorMessagesMenu.appendChild(option);
      break;
    default:
      generalEffectsMenu.appendChild(option);
      break;
  }
}
```

Reply:
Yes, I think this looks much nicer.

Reply:
done